File: nivel-02/ex-04/conversor_temp.c
#include <stdio.h>
#include <assert.h>

//
// Converte fahrenheit em celsius
//
float toCelsius(float fahrenheit) {
	float celsius = 5*((fahrenheit-32)/9);
	return celsius;
}

//
// Converte celsius em fahrenheit
//
float toFahrenheit(float celsius) {
	float fahrenheit = (celsius * 9/5) + 32;
	return fahrenheit;
}

//
// Testes
//
int main (){

    float celsius    = 100;
    float fahrenheit = 212;

    assert(celsius == toCelsius(fahrenheit));
    assert(fahrenheit == toFahrenheit(celsius));

    return 0;
}

File: nivel-02/ex-04/conversor_temp.js
var assert = require('assert');


//
// Converte fahrenheit em celsius
//
function toCelsius(fahrenheit){
	c = ((fahrenheit-32)/9)*5
	return c

}

//
// Converte celsius em fahrenheit
//
function toFahrenheit(celsius) {
	f = (celsius*9/5) + 32
	return f

}

//
// Testes
//
try {
    var celsius    = 100
    var fahrenheit = 212

    assert.equal(celsius, toCelsius(fahrenheit), "msg");
    assert.equal(fahrenheit, toFahrenheit(celsius), "msg");
} catch(e) {
    console.log(e);
}